src/core/io/platform/f3xx/ADCf3xx.cpp

```diff
@@ -137,60 +136,70 @@ void ADCf3xx::addChannel(uint8_t rank) {
         break;
     case Pin::PA_1:
-        adcChannel.Channel = ADC_CHANNEL_2;
+        channelStruct = {.adc1 = 1, .channel = static_cast<uint8_t>(ADC_CHANNEL_2)};
         break;
     case Pin::PA_2:
-        adcChannel.Channel = ADC_CHANNEL_3;
+        channelStruct = {.adc1 = 1, .channel = static_cast<uint8_t>(ADC_CHANNEL_3)};
         break;
     case Pin::PA_3:
-        adcChannel.Channel = ADC_CHANNEL_4;
+        channelStruct = {.adc1 = 1, .channel = static_cast<uint8_t>(ADC_CHANNEL_4)};
         break;
     case Pin::PA_4:
-        adcChannel.Channel = ADC_CHANNEL_5;
+        channelStruct = {.adc1 = 1, .channel = static_cast<uint8_t>(ADC_CHANNEL_5)};
         break;
     case Pin::PC_0:
-        adcChannel.Channel = ADC_CHANNEL_6;
+        channelStruct = {.adc1 = 1, .channel = static_cast<uint8_t>(ADC_CHANNEL_6)};
         break;
     case Pin::PC_1:
-        adcChannel.Channel = ADC_CHANNEL_7;
+        channelStruct = {.adc1 = 1, .channel = static_cast<uint8_t>(ADC_CHANNEL_7)};
         break;
     case Pin::PC_2:
-        adcChannel.Channel = ADC_CHANNEL_8;
+        channelStruct = {.adc1 = 1, .channel = static_cast<uint8_t>(ADC_CHANNEL_8)};
         break;
     case Pin::PC_3:
-        adcChannel.Channel = ADC_CHANNEL_9;
+        channelStruct = {.adc1 = 1, .channel = static_cast<uint8_t>(ADC_CHANNEL_9)};
         break;
     case Pin::PA_6:
-        adcChannel.Channel = ADC_CHANNEL_10;
+        channelStruct = {.adc1 = 1, .channel = static_cast<uint8_t>(ADC_CHANNEL_10)};
         break;
     case Pin::PB_0:
-        adcChannel.Channel = ADC_CHANNEL_11;
+        channelStruct = {.adc1 = 1, .channel = static_cast<uint8_t>(ADC_CHANNEL_11)};
         break;
     case Pin::PB_1:
-        adcChannel.Channel = ADC_CHANNEL_12;
+        channelStruct = {.adc1 = 1, .channel = static_cast<uint8_t>(ADC_CHANNEL_12)};
         break;
     case Pin::PB_13:
-        adcChannel.Channel = ADC_CHANNEL_13;
+        channelStruct = {.adc1 = 1, .channel = static_cast<uint8_t>(ADC_CHANNEL_13)};
         break;
     case Pin::PB_11:
-        adcChannel.Channel = ADC_CHANNEL_14;
+        channelStruct = {.adc1 = 1, .channel = static_cast<uint8_t>(ADC_CHANNEL_14)};
         break;
     case Pin::PA_7:
-        adcChannel.Channel = ADC_CHANNEL_15;
+        channelStruct = {.adc1 = 1, .channel = static_cast<uint8_t>(ADC_CHANNEL_15)};
         break;
     default:
+        channelStruct = {}; // sets all variables to 0
+        log::LOGGER.log(log::Logger::LogLevel::ERROR, "INVALID PIN 0x%x!!", pin);
         break; // Should never get here
     }
 
     // Subtract 1 because rank starts at 1
     channels[rank - 1] = pin;
 
+    // This checks if the pin being used supports the ADC being used
+    if (checkSupport(adcPeriph, channelStruct)) {
+        adcChannel.Channel = channelStruct.channel;
+    } else {
+        log::LOGGER.log(log::Logger::LogLevel::ERROR, "DOES NOT SUPPORT PIN 0x%x!!", pin);
```

Comment:
This should also probably stop since we are in a bad state.

Reply:
stop how? if your saying you want it to crash, me and magee talked about it in another comment and decided against hard faulting.